src/components/Sidebar/contentTree.js

```diff
@@ -1,23 +1,157 @@
 import React, { useState } from 'react';
-import ContentTreeGroup from './contentTreeGroup';
+import { css } from '@emotion/core';
+import { useTheme } from 'emotion-theming';
 import { calculateNavigation } from '../';
+import { Link } from "gatsby"
+import { Nav } from '@fluentui/react/lib/Nav';
+import { initializeIcons } from '@fluentui/react/lib/Icons';
+import emoji from '../../utils/emoji';
+import { onMobile } from '../../styles/responsive';
+
+initializeIcons();
+
+// Official Way to customize Fluent UI CSS
+const navStyles = props => {
+  const { isGroup } = props;
+  return {
+    chevronButton:[
+      // Add custom css class for override group elemnt
+      isGroup && 'is-group'
+    ]
+  }
+}
+
+// Override Fluent UI CSS (Not good way)
+const style = (theme) => css`
+  width: 280px;
+  ${onMobile} {
+    width: 100%;
+  }
+  margin-bottom: 30px;
+  margin-right: 10px;
+
+  .ms-Nav-compositeLink{
+    background: none;
+
+    a, a:visited{
+      color: ${theme.navigationSidebar.font.base};
+    }
+
+    &.is-selected button{
+      background: none;
+    }
+    &.is-selected {
+      background-color: ${theme.navigationSidebar.backgroundActive};
+    }
+    &:hover{
+      background-color: ${theme.navigationSidebar.backgroundHover};
+    }
+  }
+
+  .ms-Nav-chevronButton{
+    font-size: 16px;
+    color: ${theme.navigationSidebar.font.base};
+
+    // Not official css class, overriding from above.
+    &.is-group{
+      border-bottom: 1px solid ${theme.navigationSidebar.divider} !important;
+    }
+  }
+
+  .ms-Nav-navItem .ms-Nav-link:hover{
+    color: ${theme.navigationSidebar.font.hover}
+  }
+
+  .ms-Nav-link
+   {
+    background: none !important;
+  }
+`;
+
+const extractFirstLevelGroupId = (pathname) => {
+  const trimLastSlash = pathname.replace(/\/$/, '');
+  const trimFirstSlash = trimLastSlash.replace(/^\//, '');
+  return trimFirstSlash.substr(0, trimFirstSlash.indexOf('/')); 
+}
+
+const extractSecondLevelGroupId = (pathname) => {
+  const trimLastSlash = pathname.replace(/\/$/, '');
+  const trimFirstSlash = trimLastSlash.replace(/^\//, '');
+  const removedFirstLevel = trimFirstSlash.substr(trimFirstSlash.indexOf('/')).replace(/^\//, ''); 
+  return removedFirstLevel.substr(0, removedFirstLevel.indexOf('/')); 
+}
```

Comment:
@mildronize  Please accept my apologies if I misunderstand your code but I think we can simplify this code to something like this if I understand it correctly.
```ts
const pathName =  '/abc/xyz/'.replace(/(^\/|\/$)/g, '');
const pathSegments = pathName.split('/');

const firstLevelGroupId = pathSegments[0]
const secondLevelGroupId = pathSegments[1]

```

![image](https://user-images.githubusercontent.com/2938310/127969450-a893a3ae-a908-43ee-9456-1b7e519bf11d.png)

If it is what you want, I think you can apply it and handle some path that has only 1, 2, 3, 4.. levels or make a help function and pass an index number/level number that you want.

Reply:
Supreme Way! I'll, Thank you a lot. For me, i just to make it works first.

Reply:
```js
const extractLevelGroupID = pathname => pathname.replace(/(^\/|\/$)/g, '').split('/');
```

For now, one line function.